### PR TITLE
feat: add dynamic tables to sdk

### PR DESCRIPTION
**Pull request by sfc-gh-swinkler**
[SNOW-871686]
Adds dynamic tables to SDK

### CODE REVIEW

File: pkg/sdk/dynamic_table.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	if dtr.RefreshModeReason.Valid {
+		dt.RefreshModeReason = dtr.RefreshModeReason.String
+	}
+	if dtr.LastSuspendedOn.Valid {
+		dt.LastSuspendedOn = dtr.LastSuspendedOn.Time
+	}
+	return dt
+}
+
+func (dt *dynamicTables) Show(ctx context.Context, opts *ShowDynamicTableOptions) ([]*DynamicTable, error) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/dynamic_table.go:165]: We put `Show` methods to, in this case, `dynamic_table_impl.go` along with other methods.

File: pkg/sdk/dynamic_table.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	if opts == nil {
+		opts = &ShowDynamicTableOptions{}
+	}
+	if err := opts.validate(); err != nil {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/dynamic_table.go:169]: Please consult e.g. `database_role_impl.go`. You can simplify code in this method by using our helper `validateAndQuery` and `convertRows` functions (the second one requires you to change `toDynamicTable` to `convert`).

File: pkg/sdk/dynamic_table_test.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
+package sdk
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDynamicTableCreate(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/dynamic_table_test.go:10]: We try to put more validation tests on unit test level, consult e.g. `database_role_test.go`.

File: pkg/sdk/dynamic_table_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt_DynamicTableCreate(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/dynamic_table_integration_test.go:11]: Please add tests for alter and drop

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/sdk/dynamic_table.go (hunk 1 above)
Comment (sfc-gh-swinkler) [reply pkg/sdk/dynamic_table.go:165]: done

File: pkg/sdk/dynamic_table.go (hunk 2 above)
Comment (sfc-gh-swinkler) [reply pkg/sdk/dynamic_table.go:169]: done

File: pkg/sdk/dynamic_table_test.go (hunk 3 above)
Comment (sfc-gh-swinkler) [reply pkg/sdk/dynamic_table_test.go:10]: done

File: pkg/sdk/dynamic_table_integration_test.go (hunk 4 above)
Comment (sfc-gh-swinkler) [reply pkg/sdk/dynamic_table_integration_test.go:11]: added for alter, i am not sure if we need one for drop. i have renamed the create to "TestInt_DynamicTableCreateAndDrop" to make it more clear

**Review by sfc-gh-asawicki — APPROVED** [review]

File: pkg/sdk/helper_test.go [hunk 5]
```diff
@@ -711,6 +717,56 @@ func createStage(t *testing.T, client *Client, database *Database, schema *Schem
...
+		}
+		if schemaCleanup != nil {
+			schemaCleanup()
+		}
+		if databaseCleanup != nil {
+			databaseCleanup()
+		}
+		if warehouseCleanup != nil {
+			warehouseCleanup()
+		}
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/helper_test.go:768]: Is something wrong with this file? It looks like you cut out closing `}` ?

Comment (sfc-gh-swinkler) [reply pkg/sdk/helper_test.go:768]: yes you are right haha. fixed.

Comment (sfc-gh-asawicki) [reply pkg/sdk/helper_test.go:768]: Well, it seems it's still wrong :(

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/sdk/helper_test.go (hunk 5 above)
Comment (sfc-gh-swinkler) [reply pkg/sdk/helper_test.go:768]: apparently i did not push my commit 😮

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-swinkler — COMMENTED** [review]
👍

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**